packages/datagateway-download/src/downloadStatus/downloadStatusTable.component.test.tsx

```diff
@@ -369,4 +369,52 @@ describe('Download Status Table', () => {
       }
     });
   });
+
+  it('should refresh download progress when refresh button is clicked', async () => {
+    (
+      getPercentageComplete as jest.MockedFunction<typeof getPercentageComplete>
+    ).mockResolvedValue(30);
+
+    renderComponent({
+      settings: {
+        ...mockedSettings,
+        uiFeatures: {
+          downloadProgress: true,
+        },
+      },
+    });
+
+    await waitFor(() => {
+      for (const progressBar of screen.getAllByRole('progressbar')) {
+        expect(progressBar).toBeInTheDocument();
+      }
+      for (const progressText of screen.getAllByText('30%')) {
+        expect(progressText).toBeInTheDocument();
+      }
+    });
+
+    // pretend the server returns an updated value
+    (
+      getPercentageComplete as jest.MockedFunction<typeof getPercentageComplete>
+    ).mockResolvedValue(50);
+
+    // pretend table is refreshed
+    renderComponent({
+      settings: {
+        ...mockedSettings,
+        uiFeatures: {
+          downloadProgress: true,
+        },
+      },
+    });
```

Comment:
This test isn't testing the refresh button behaviour? Instead this is just rendering a new table, which is why the next assertation passes. Also, there's already a unit test for refresh behaviour - we should piggy back that test to test the progress bars as well. I'll commit the changes as I was fiddling with the code to make sure I was right that this test wasn't triggering the refresh behaviour.

Reply:
Yeh I was struggling with that yesterday as the refresh button for the user table is in a separate component I think so couldn't be pressed, but you are right I think the way I wrote the test makes it redundant. Thanks for the changes having the test there makes more sense